client/my-sites/navigation/index.jsx

```diff
@@ -18,6 +20,15 @@ class MySitesNavigation extends React.Component {
 		event.stopPropagation();
 	};
 
+	renderSidebar() {
+		if ( 'jetpack-cloud' === config( 'sidebar' ) ) {
```

Comment:
I really think we shouldn't be doing this and should be setting up our sidebar from the controller. The layout should then be able to render it. We don't benefit anything by reusing this component and throwing away 100% of what it's doing. That being said, I'm suggesting to reconsider what controller functionality we reuse in Jetpack Cloud, and to reuse only what we really use.

Reply:
@tyxla I wasn't totally sure if I did what you intended, but check out the changes I pushed and see. I think if that is what you were going for then it might be worth doing something similar for the masterbar.